pkg/backup/backup_job.go

```diff
@@ -1899,15 +1903,36 @@ func (b *backupResumer) maybeNotifyScheduledJobCompletion(
 			// Not a scheduled backup.
 			return nil
 		}
-
-		scheduleID := jobspb.ScheduleID(tree.MustBeDInt(datums[0]))
+		scheduleID = jobspb.ScheduleID(tree.MustBeDInt(datums[0]))
 		if err := jobs.NotifyJobTermination(ctx, txn, env, b.job.ID(), jobState, b.job.Details(), scheduleID); err != nil {
 			return errors.Wrapf(err,
 				"failed to notify schedule %d of completion of job %d", scheduleID, b.job.ID())
 		}
 		return nil
-	})
-	return err
+	}); err != nil {
+		return err
+	}
+
+	if scheduleID != 0 {
```

Comment:
we should add another cluster setting along the lines of `compaction.enabled` and check it here. If we add this cluster setting, we can have default chain length of 3.